thorlcr/thorutil/thmem.cpp

```diff
@@ -646,7 +662,17 @@ void CThorExpandingRowArray::compact()
         }
     }
     numRows = freeFinger-rows;
-    // JCSMORE - this may be a good time to call IRowManager::compactRows
+    memsize_t numEmptiedPages = rowManager->compactRows(numRows, rows);
+
+#ifdef _DEBUG
+    ActPrintLog(&activity, "CThorExpandingRowArray::compact(): compactRows freed %"I64F"d pages", (unsigned __int64)numEmptiedPages);
+#endif
+
+    if (maxSpillCost)
```

Comment:
Is this test correct?  If this always shrinks, then it should be possible to resize the row array without causing any spills.  Should the parameter be named differently?  Should 0 always be passed to resize?